python/test/unit/language/test_core.py

```diff
@@ -2932,6 +2932,32 @@ def _kernel(out, ALLOW_TF32: tl.constexpr):
     assert torch.all(out == out_ref)
 
 
+def test_dot_with_transposed_arg(device):
+    M = N = K = 32
+    in1 = torch.ones((M, K), dtype=torch.float32, device=device)
+    in2 = torch.ones((K, N), dtype=torch.float32, device=device)
+    out = torch.zeros((M, N), dtype=torch.float32, device=device)
+
+    @triton.jit
+    def _kernel(in1_ptr, in2_ptr, output_ptr, in_stride, in2_stride, out_stride,
+                M: tl.constexpr, N: tl.constexpr, K: tl.constexpr):
+        M_offsets = tl.arange(0, M)
+        N_offsets = tl.arange(0, N)
+        K_offsets = tl.arange(0, K)
+        in_offsets = M_offsets[:, None] * in_stride + K_offsets[None, :]
+        in2_offsets = K_offsets[:, None] * in2_stride + N_offsets[None, :]
+        x = tl.load(in1_ptr + in_offsets, mask=in_offsets < M * K)
+        w = tl.load(in2_ptr + in2_offsets, mask=in2_offsets < K * N)
+        o = tl.dot(x, w.T, out_dtype=tl.float32)
+        output_offsets = M_offsets[:, None] * out_stride + N_offsets[None, :]
+        tl.store(output_ptr + output_offsets, o, mask=output_offsets < M * N)
+
+    _kernel[(1, )](in1, in2, out, in1.stride()[0], in2.stride()[0], out.stride()[0], M=M, N=N, K=K)
+
+    reference_out = torch.matmul(in1, in2)
```

Comment:
In this case, it doesn't matter whether in2 is transposed or not since all values are 1s. Could you please make it a bit more complicated? Like using random values to initialize in2?